yellowbox/extras/redis.py

```diff
@@ -1,17 +1,30 @@
-from typing import TypeVar, Callable
+from contextlib import contextmanager
+from typing import TypeVar, Callable, Mapping, Union, Sequence
 
 from docker import DockerClient
 from redis import ConnectionError as RedisConnectionError, Redis
 
 from yellowbox.containers import get_ports, create_and_pull
-from yellowbox.subclasses import SingleContainerService, RunnableWithContext
+from yellowbox.subclasses import SingleContainerService, RunMixin
 from yellowbox.utils import retry
 
 REDIS_DEFAULT_PORT = 6379
 _T = TypeVar("_T")
+RedisPrimitive = Union[str, int, float, bytes]
+RedisState = Mapping[str, Union[RedisPrimitive, Mapping[str, RedisPrimitive], Sequence[RedisPrimitive]]]
 
 
-class RedisService(SingleContainerService, RunnableWithContext):
+def append_state(client: Redis, db_state: RedisState):
```

Comment:
Rethinking - This does not append the state. It might be true on rpush but totally different on set and hset which replace...